### ISSUE TITLE
Possibly undefined value not caught when getting properties via interface index signatures

### ISSUE THREAD

**Issue by sdrsdr** [microsoft/TypeScript#35265]
**TypeScript 3.7.2**
[Playground link](http://www.typescriptlang.org/play/#code/JYOwLgpgTgZghgYwgAgCZzHA+hAHmKRMAeyixmQG8AoASAAoBbAZwHMAuZOEATwEp2AI2LEANhG4BuagF9q1UJFiIU6TDnyEEJKMyxgqdANpqMPAA4R2zAqFYBddqY0EipctLnVxB53lfapJx+mm66+sgAvFQy0tQhATpGAOTMwIzm4sn2kTAAriDawMQgyPTILBxcvALCYhKlNLRQEGB5UKXwoswQnvKK0PBIyP5aOlh5PVDIhrSjYU4Y2POBZDB93q0jUFDu7iDEYOwr45PQUTQzMydBaEsuY6QpBQDWBwDuINmy8tC7ZPtDgA6G5QegAIhecDqjCBcPBfEkQA)

Compiler Options:
```json
{
  "compilerOptions": {
    "noImplicitAny": true,
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "strictPropertyInitialization": true,
    "strictBindCallApply": true,
    "noImplicitThis": true,
    "noImplicitReturns": true,
    "useDefineForClassFields": false,
    "alwaysStrict": true,
    "allowUnreachableCode": false,
    "allowUnusedLabels": false,
    "downlevelIteration": false,
    "noEmitHelpers": false,
    "noLib": false,
    "noStrictGenericChecks": false,
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "esModuleInterop": true,
    "preserveConstEnums": false,
    "removeComments": false,
    "skipLibCheck": false,
    "checkJs": false,
    "allowJs": false,
    "experimentalDecorators": false,
    "emitDecoratorMetadata": false,
    "target": "ES2017",
    "module": "ESNext"
  }
}
```

**Input:**
```typescript
interface data_extractor_f {
	(msg: any):boolean;
}

interface data_extractors_t {
	[datatype:string]:data_extractor_f;
}

let data_extractor: data_extractors_t = {};

data_extractor['simple']=function ( msg: any):boolean {
	return false;
}

interface extractor_user  {
	extractor:data_extractor_f;
}

let error_or_not:extractor_user ={
    extractor: data_extractor['unknown']
}

error_or_not.extractor("kaboom...");
```

**Output:**
```javascript
"use strict";
let data_extractor = {};
data_extractor['simple'] = function (msg) {
    return false;
};
let error_or_not = {
    extractor: data_extractor['unknown']
};
error_or_not.extractor("kaboom...");

```

**Expected behavior:**
IMHO compiler should exit with error on 
```
 extractor: data_extractor['unknown'] 
```
saying type `data_extractor_f|undefined` is not assignable to  `data_extractor_f` as the compiler has no way to verify if the index is actually defined in the object.

**Comment by sdrsdr** [comment]
Ok if you do
```
interface data_extractors_t {
	[datatype:string]:data_extractor_f|undefined;
}
```
it now complains about the probably undefined value but shouldn't this be the default behavior?

**Comment by MartinJohns** [comment]
Duplicate of #13778.

**Comment by sdrsdr** [comment]
so .. my bad .. but option to force this might be nice .. or a warning on any index signature not including | undefined;
